src/Dsn.php

```diff
@@ -99,7 +99,7 @@ public static function createFromString(string $value): self
         $segmentPaths = explode('/', $parsedDsn['path']);
         $projectId = array_pop($segmentPaths);
 
-        if ((int) $projectId <= 0) {
+        if (!ctype_digit($projectId)) {
```

Comment:
This will still fail if the project ID is a non-number, so project ID cannot be a UUID for example with this change still.

Shouldn't we move this check to the getter when returning `int` because returning a non-number as `int` could cause unexpected results (which is probably why this check is still here)?

Reply:
>This will still fail if the project ID is a non-number, so project ID cannot be a UUID for example with this change still.

You're right, but this is what other SDK still do, e.g. [here](https://github.com/getsentry/sentry-javascript/blob/f8aeeeab0006c9a0527fd8392bede8c534648325/packages/utils/src/dsn.ts#L50) and [here](https://github.com/getsentry/sentry-python/blob/b63b5ae32d254771cbd027009d0b6def7e2e07fb/sentry_sdk/utils.py#L188). I understand the concern that if you pass something that is not a number there will still be issues, but there are no public information that I'm aware of that the IDs will not be just digits also in the future

>returning a non-number as `int` could cause unexpected results

I wouldn't worry too much about the the cast in the getter because in the worst case the project ID will be `0`

Reply:
Check the original issue. It's stated that it needs to be opaque, and UUID is mentioned explicitly too 😬

Reply:
Well, it's written that it could be an UUID or a 64bit number. I have no problems in removing this check, but since all other SDK seems to still require a number (stringified), I wonder if we should do it or not. @smeubank what do you suggest to do?

Reply:
@HazAT can i get a PHP sanity check here, so to comply with the original request it needs to be UUID or 64 bit int?